lib/elm-test.js

```diff
@@ -1,193 +1,77 @@
 // @flow
 
-var packageInfo = require('../package.json');
-var chalk = require('chalk');
-var Install = require('./Install.js');
-var Compile = require('./Compile.js');
-var Generate = require('./Generate.js');
-var processTitle = 'elm-test';
-var which = require('which');
-
-process.title = processTitle;
-
-function clearConsole() {
-  process.stdout.write(
-    process.platform === 'win32' ? '\x1B[2J\x1B[0f' : '\x1B[2J\x1B[3J\x1B[H'
-  );
-}
-
-process.on('uncaughtException', function (error) {
-  if (/ an argument in Javascript/.test(error)) {
-    // Handle arg mismatch between js and elm code. Expected message from Elm:
-    // "You are giving module `Main` an argument in JavaScript.
-    // This module does not take arguments though! You probably need to change the
-    // initialization code to something like `Elm.Test.Generated.Main.fullscreen()`]"
-    console.error('Error starting the node-test-runner.');
-    console.error(
-      "Please check your Javascript 'elm-test' and Elm 'node-test-runner' package versions are compatible"
-    );
-    process.exit(1);
-  } else {
-    console.error('Unhandled exception while running the tests:', error);
-    process.exit(1);
-  }
-});
-
-var fs = require('fs-extra'),
-  os = require('os'),
-  glob = require('glob'),
-  path = require('path'),
-  minimist = require('minimist'),
-  chokidar = require('chokidar'),
-  Runner = require('./Runner.js'),
-  Supervisor = require('./Supervisor.js');
-
-// Check Node.js version.
-const nodeVersionMin = '10.13.0';
-const nodeVersionString = process.versions.node;
-
-if (
-  nodeVersionString.localeCompare(nodeVersionMin, 'en', { numeric: true }) < 0
-) {
-  console.error(`You are using Node.js v${nodeVersionString}.`);
-  console.error(
-    `elm-test requires Node.js v${nodeVersionMin} or greater - upgrade the installed version of Node.js and try again!`
-  );
-  process.exit(1);
-}
-
-var args = minimist(process.argv.slice(2), {
-  boolean: ['warn', 'version', 'help', 'watch'],
-  string: ['compiler', 'seed', 'report', 'fuzz'],
-});
-var processes = Math.max(1, os.cpus().length);
+const chalk = require('chalk');
+const chokidar = require('chokidar');
+const fs = require('fs-extra');
+const glob = require('glob');
+const os = require('os');
+const path = require('path');
+const which = require('which');
+const packageInfo = require('../package.json');
+const Compile = require('./Compile.js');
+const Flags = require('./Flags.js');
+const Generate = require('./Generate.js');
+const Install = require('./Install.js');
+const Report = require('./Report.js');
+const Runner = require('./Runner.js');
+const Supervisor = require('./Supervisor.js');
+
+void Report;
 
 function flatMap(array, f) {
   return array.reduce((result, item) => result.concat(f(item)), []);
 }
 
 // Recursively search directories for *.elm files, excluding elm-stuff/
 function resolveFilePath(filename) {
```

Comment:
```suggestion
function resolveFilePath(elmFilePathOrDir) {
```

Tangential to this PR so feel free to ignore.